src/odin/maneuversbuilder.cc

```diff
@@ -3019,6 +3024,31 @@ void ManeuversBuilder::MatchGuidanceViewJunctions(Maneuver& maneuver,
   }
 }
 
+void ManeuversBuilder::ProcessGuidanceViewSignBoards(Maneuver& maneuver) {
+  // Loop over edges
+  uint32_t edge_count = 0;
+  for (uint32_t node_index = maneuver.begin_node_index();
+       ((node_index < maneuver.end_node_index()) && (edge_count < kOverlaySignBoardEdgeMax));
+       ++node_index, edge_count++) {
+    // Loop over guidance view signboards
+    auto curr_edge = trip_path_->GetCurrEdge(node_index);
+    if (curr_edge && (curr_edge->has_sign())) {
+      // Process overlay guidance view signboards
+      for (const auto& base_guidance_view_signboard : curr_edge->sign().guidance_view_signboards()) {
+        auto base_tokens = split(base_guidance_view_signboard.text(), ';');
+        // If base(is_route_number) guidance view board and a pair...
+        if (base_guidance_view_signboard.is_route_number() && is_pair(base_tokens)) {
```

Comment:
Is this check needed given that signboards never have overlays?

Reply:
yes.  base has a route number and there should be a pair